packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1144,12 +1229,41 @@ class SliverMultiBoxAdaptorElement extends RenderObjectElement implements Render
     _didUnderflow = value;
   }
 
+  /// Check if the given child is actually in childList.
+  ///
+  /// when the keep alive child is garbage collected due to
+  /// scrolling out of view, it will store in _keepAliveBucket and
+  /// not included in the childList.
+  bool _isChildInChildList(RenderObject child){
+    if(child != null){
+      final SliverMultiBoxAdaptorParentData childParentData = child.parentData;
+      if (childParentData != null){
+        return !childParentData.keptAlive;
+      }
+    }
+    return false;
+  }
+
+  RenderObject _findAfterBySlot(RenderObject child, int slot) {
+    for (int i = slot; i>= 0; i--) {
```

Comment:
Now that we do the updates in order again I am wondering if we can go back to using `_currentBeforeChild`, which may be more efficient than iterating here?

Reply:
removed